devito/stencilkernel.py

```diff
@@ -127,10 +128,22 @@ def arguments(self, *args, **kwargs):
         arguments = OrderedDict([(arg.name, arg) for arg in self.parameters])
         dim_sizes = {}
 
+        # Have we been provided substitutes for symbol data?
+        # Only SymbolicData can be overridden with this route
+        r_args = [f_n for f_n, f in arguments.items() if isinstance(f, SymbolicData)]
```

Comment:
The objects here that are not `SymbolicData` are `BufferedDimension` objects. These objects don't have this property. This leads to `AttributeError: 'BufferedDimension' object has no attribute 'is_SymbolicData'`